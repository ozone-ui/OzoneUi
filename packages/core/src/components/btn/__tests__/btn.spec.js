import { describe, it, expect } from 'vitest'
import { shallowMount, mount } from '@vue/test-utils'
import { Btn } from './'

describe('Button', () => {
  it('renders correct text', () => {
    const wrapper = mount(Btn, {
      props: { text: 'test' },
      slots: { default: 'test' },
    })
    console.log(wrapper.classes())
    expect(wrapper.text()).toBe('test')
  })
})
it('provides correct classes for default color button', () => {
  const defaultButtonClasses = [
    'px-[12px]',
    'py-[6px]',
    'rounded-[8px]',
    'font-bold',
    'text-[14px]',
    'flex',
    'items-center',
    'justify-center',
    'gap-x-[8px]',
    'hover:brightness-110',
    'transition-all',
    'duration-200',
    'w-fit',
  ]

  const wrapper = mount(Btn, { props: { color: '#fff' } })
  const classes = wrapper.classes()

  defaultButtonClasses.forEach((cl) => expect(classes).toContain(cl))
})
