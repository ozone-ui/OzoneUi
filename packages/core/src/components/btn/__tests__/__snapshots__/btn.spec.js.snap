// Vitest Snapshot v1

exports[`matches snapshot 1`] = `"<button class=\\"px-[12px] py-[6px] rounded-[8px]  font-bold text-[14px] flex items-center justify-center gap-x-[8px] hover:brightness-110 transition-all duration-200  w-fit\\" style=\\"background: red; color: rgb(0, 0, 0); border-color: transparent; cursor: pointer;\\" target=\\"_blank\\">test</button>"`;
